test/pylib/scylla_cluster.py

```diff
@@ -680,6 +680,9 @@ class ScyllaClusterManager:
 
     def __init__(self, test_uname: str, clusters: Pool[ScyllaCluster], base_dir: str) -> None:
         self.test_uname: str = test_uname
+        # The currently running test case with self.test_uname prepended, e.g.
+        # test_topology.1::test_add_server_add_column
+        self.current_test_case_full_name: str = ''
```

Comment:
I think you have too many concepts of test names: You mentioned "test name", "test case name", "unique name" and now "current_test_case_full_name" which contains pieces of all the previous ones. None of these variable names are self-explanatory at all (is "full" more full than not-full? Is unique more unique than not-unique? What's the difference between a test and a test case?).

In my opinion (but feel free to disagree), not all of these should be called a test "name" or even exist. I think there is just **one** concept of **test name**, the pytest concept (e.g., `test_topology::test_add_server_add_column`). To that you can add a concept of "run number" (not a name - a number. e.g, the 3rd run of the same test, or the 234th test ran today, however you want to number test runs), and then you can have "test_name_and_number" which includes both.

Reply:
`test.py` uses the following:
- "test name" to name an entire test file (which contains multiple test functions),
- "test uname" (i.e. "test unique name") to name a single run of a test file in the context of `--repeat`.

There is currently no way to run a single test function using `test.py`.

In scylla_cluster.py we continued to use "test name" to refer to test files, but then "test name" was also used to refer to the test cases/functions. This was confusing. **The multiple concepts were already there, I only gave them different names**. So I decided to use "test case" to name a single test function within test file. I think it's reasonable.

Reply:
Since what you want to print is the test *function* name, and just the test file name is never interesting, I don't understand why these test.py-invented concepts are productive. You already had existing code which had for "test name"  somefile.py::sometestfunction - why change the name of this variable (to "test_case")?

But here, it's not just my annoyance with changing the pre-existing pytest terminology. As I noted you end up with *four* different concepts of "test name". I think this is a bit too much. Next month not even you will remember which variable says what - what is the "unique", what is the "full" name, and the "test case" name, which of these includes just the file name or also the function name, and so on.

My primary claim is that I believe you don't need *any* variable which tracks just the test file name, without the test function name. I think such a variable is useless. If test.py forces you to take such a variable (i.e. unique_name contains the test file name plus some extra suffix) think you should quickly extract what you really want from this variable (the extra suffix) and then use only that suffix as a separate variable - not a 4th variant of "test name".

Reply:
> You already had existing code which had for "test name" somefile.py::sometestfunction - why change the name of this variable (to "test_case")?

There was no variable which stored `somefile.py::sometestfunction`. There was either `somefile` or `sometestfunction`. And in both these cases, a variable named `test_name` was used. I wanted to distinguish one from the other. For the case of `somefile`, I decided to continue using `test_name`, because in `test.py` "test" refers to a test file. So we need a name for `sometestfunction` - and I picked "test case".

But then I wanted to print `somefile.1::sometestfunction`. I needed a name for this thing. I decided to use "full test case name", where "full" means we print the entire context (so the test file from which this case comes, and the unique repeat number).

Naming stuff is hard.

> As I noted you end up with four different concepts of "test name". I think this is a bit too much. Next month not even you will remember which variable says what - what is the "unique", what is the "full" name, and the "test case" name, which of these includes just the file name or also the function name, and so on.

These **are** different concepts, that's a fact, not a choice I made. The only choice I made is to give different names to different concepts.
At least now it's clear that these things are different just from reading the code.
Before you only had `test_name` and you didn't know if it refers to `somefile` or `sometestfunction`, it could mean either.

> My primary claim is that I believe you don't need any variable which tracks just the test file name, without the test function name.

We do. That's what `test.py` gives us when it initializes `ScyllaClusterManager`. We only learn about the test case name when a test uses the `manager` fixture which calls `before_test` which passes the test case name through a REST API to `ScyllaClusterManager`.

Reply:
> > My primary claim is that I believe you don't need any variable which tracks just the test file name, without the test function name.
> 
> We do. That's what `test.py` gives us when it initializes `ScyllaClusterManager`. We only learn about the test case name when a test uses the `manager` fixture which calls `before_test` which passes the test case name through a REST API to `ScyllaClusterManager`.

I give up. In my opinion, test.py and test/pylib/* have become too big and too intertwined. Both code and terminology are leaking between the two, to an unnecessary degree. But as long as existing test suites like test/alternator (right now, 87 times more testing code than test/topology!) and test/cql-pytest (100 times larger than test/toplogy) are not being negatively impacted, I can live with that.

I'll commit your PR.